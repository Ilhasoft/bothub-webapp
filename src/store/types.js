export default {
  SET_TOKEN: 'SET_TOKEN',
  OPEN_LOGIN_MODAL: 'OPEN_LOGIN_MODAL',
  CLOSE_LOGIN_MODAL: 'CLOSE_LOGIN_MODAL',
  SET_PROFILE: 'SET_PROFILE',
  SET_NICKNAME_AUTHENTICATED: 'SET_NICKNAME_AUTHENTICATED',
  SET_REPOSITORY_RELATED_UUID: 'SET_REPOSITORY_RELATED_UUID',
  CACHE_FETCH: 'CACHE_FETCH',
};
