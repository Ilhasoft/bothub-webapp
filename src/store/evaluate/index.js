import actions from './actions';
import state from './state';
import mutations from './mutations';
import getters from './getters';
import types from './types';

export default {
  actions,
  state,
  mutations,
  getters,
  types,
};
