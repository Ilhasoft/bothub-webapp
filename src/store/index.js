import Vue from 'vue';
import Vuex from 'vuex';

import Auth from './auth';
import User from './user';
import Org from './org';
import Repository from './repository';
import Category from './category';
import Example from './example';
import EvaluateExample from './evaluate-example';
import Translate from './translate';
import CachedFetch from './cached-fetch';
import Entity from './entity';
import Intent from './intent';
import News from './news';
import Tutorial from './tutorial';
import External from './external';

Vue.use(Vuex);

const store = new Vuex.Store({
  modules: {
    Auth,
    User,
    Org,
    Repository,
    Category,
    Example,
    Translate,
    CachedFetch,
    EvaluateExample,
    Entity,
    News,
    Tutorial,
    External,
    Intent,
  },
});

store.dispatch('retriveAuthToken');

export default store;
