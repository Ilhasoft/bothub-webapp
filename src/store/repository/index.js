import state from './state';
import getters from './getters';
import actions from './actions';
import mutations from './mutations';
import types from './types';

export default {
  state,
  getters,
  actions,
  mutations,
  types,
};
