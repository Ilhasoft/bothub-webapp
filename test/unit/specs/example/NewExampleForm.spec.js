/* eslint-disable import/first */
jest.mock('@/api/request');

import { shallowMount, createLocalVue } from '@vue/test-utils';
import BH from 'bh';
import store from '@/store';
import NewExampleForm from '@/components/example/NewExampleForm';


const localVue = createLocalVue();
localVue.use(BH);

describe('NewExampleForm.vue', () => {
  let wrapper;
  beforeEach(() => {
    store.replaceState({
      Auth: {},
    });
    wrapper = shallowMount(NewExampleForm, {
      localVue,
      propsData: {
        repository: {
          uuid: '8511fd26-a3bc-4f74-9af1-176abca5401d',
          language: 'en',
        },
      },
      store,
    });
  });

  test('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('set text and intent, add entity', () => {
    const text = 'my name is douglas';
    const intent = 'greet';
    const entity = {
      start: 11,
      end: 18,
      entity: 'name',
    };
    beforeEach(() => {
      wrapper.vm.text = text;
      wrapper.vm.intent = intent;
      wrapper.vm.addNewEntity(entity);
    });

    test('check data', () => {
      const { data } = wrapper.vm;

      expect(data.text).toBe(text);
      expect(data.intent).toBe(intent);
      expect(data.entities[0]).toMatchObject(entity);
    });
  });
});
