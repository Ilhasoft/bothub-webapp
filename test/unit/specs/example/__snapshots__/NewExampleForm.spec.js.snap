// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewExampleForm.vue renders correctly 1`] = `
<div>
  <div class="columns is-variable is-1">
    <div class="column">
      <form>
        <div class="columns is-variable is-2">
          <div class="column is-three-fifths">
            <bhfield-stub>
              <exampletextwithhighlightedentitiesinput-stub size="medium" placeholder="Add a sentence" formatters="function (v) {
      /* istanbul ignore next */cov_1oyjq7f3e2.f[1]++;
      cov_1oyjq7f3e2.s[3]++;
      return v.trimStart();
    },function (v) {
      /* istanbul ignore next */cov_1oyjq7f3e2.f[3]++;
      cov_1oyjq7f3e2.s[5]++;
      return v.replace('\\n', '');
    },function (v) {
      /* istanbul ignore next */cov_1oyjq7f3e2.f[5]++;
      cov_1oyjq7f3e2.s[7]++;
      return v.replace(/\\s{2,}/g, ' ');
    }"></exampletextwithhighlightedentitiesinput-stub>
            </bhfield-stub>
          </div>
          <div class="column">
            <bhfield-stub>
              <bhautocomplete-stub size="medium" placeholder="Intent" formatters="function (v) {
      /* istanbul ignore next */cov_2522u8mndo.f[10]++;
      cov_2522u8mndo.s[26]++;
      return v.toLowerCase().replace(' ', '_').replace(/[,./\\\\;+=!?@#$%¨&amp;*()[\\]^&quot;'~{}ç:<>|]/, '');
    }"></bhautocomplete-stub>
            </bhfield-stub>
          </div>
          <div class="column is-narrow">
            <bhfield-stub>
              <bhbutton-stub>Submit</bhbutton-stub>
            </bhfield-stub>
          </div>
        </div>
      </form>
    </div>
  </div>
  <bhfield-stub></bhfield-stub>
  <div class="columns is-variable is-1">
    <div class="column is-narrow">
      <newentityinput-stub></newentityinput-stub>
    </div>
  </div>
</div>
`;
